### ISSUE TITLE
Reduce use of **kwargs — stop silently swallowing sorry specifiers

### ISSUE THREAD

**Issue by gerritholl** [pytroll/satpy#1277]
## Feature Request

**Is your feature request related to a problem? Please describe.**

Satpy (and other Pytroll projects) use `**kwargs` a lot.  This can be useful, but it's a problem when typos in keyword arguments fail to raise an exception:

```python
sc.save_datasets(filenane="/tmp/where-is-my-file.png")
sc.resample(resempler="native")
```

Neither of those raises an exception or even a warning.

**Describe the solution you'd like**

Somewhere down the call stack, a function to which `**kwargs` is passed on should _not_ accept arbitrary keyword arguments, or should ensure that they are exhaused and raise a `TypeError` manually otherwise.

**Describe any changes to existing user workflow**

It should not affect working code, but it may cause currently buggy code to start raising errors instead of failing silently.

**Additional context**

**Comment by djhoese** [comment]
I think the main issue here is what we've been discussing on slack the last week or so: Satpy breaks Single Responsibility Priniciple (https://en.wikipedia.org/wiki/Single-responsibility_principle) a lot by having objects/functions that do *a lot*. The other cause is when we call some other pytroll library (pycoast, pydecorate, pyresample, etc) where the wrapping satpy function has no other logical choice than to use `**kwargs` (otherwise we might limit Satpy's functionality just because of what kwargs we can pass. I guess we could specify all the ones we know and any new ones can be passed by `**kwargs` but that doesn't stop mistyped kwargs from getting through.

One solution that would help is embracing SRP a little more and requiring the user to perhaps call a few extra methods/functions in their work flow (ex. split save_dataset into enhance, add_overlay, add_decoration, save). We could also improve other pytroll libraries to better support xarray objects so we can call directly into these libraries instead of wrapping them in some satpy function.

**Comment by mraspaud** [comment]
I agree with what is said here. Hopefully we will start an refactoring effort soon that will help solve some of these issues.
